Automattic/jetpack
# Add an i18n framework from Calypso.

**_Copy of @zinigor's branch from jetpack-react_**

This PR adds Calypso as npm dependency and borrows some of Calypso's functionality in order to provide i18n capabilities to client-side code.

- [x] Add Calypso as dependency.
- [x] Create a build script that will import strings from JS into a PHP file in order to translate them.
- [x] Add a mechanism that will export strings from PHP to JS in order enable the i18n mixin to use them.

To test:

- run `npm build` to generate necessary files in dependencies.
- use the plugin as usual, it should work properly.
- try to `import { translate as __ } from 'lib/mixins/i18n'` in your module.
- try to use `__( 'Translatable string' )` instead of `Translatable string` in your code.
- try to switch your WordPress to a different language and see if [already translated strings](https://translate.wordpress.org/projects/wp-plugins/jetpack/dev/es/default) appear properly translated. 
- bonus points if you try to use placeholders and plural versions of strings, you can read more about that in [the mixin docs](https://github.com/Automattic/wp-calypso/tree/master/client/lib/mixins/i18n).
- extra bonus points if you use `__` for your module and run `npm build` again. You should see the newly generated `jetpack-strings` file with your newly added strings.
